h2o-algos/src/main/java/hex/util/LinearAlgebraUtils.java

```diff
@@ -19,6 +19,16 @@
 
 public class LinearAlgebraUtils {
   /*
+   * generate a frame of zeros with number of rows and columns defined by rowCount, colCount
+   */
+  public static Frame generateFrameOfZeros(int rowCount, int colCount) {
+    assert (rowCount > 0) && (colCount > 0);  // make sure we have positive row and column numbers
+
```

Comment:
I think you are leaking the tempVec here. It has to be re-used in the returned frame or removed before you exit the method.